api/nodes/serializers.py

```diff
@@ -492,7 +492,10 @@ def update(self, instance, validated_data):
                 raise exceptions.NotFound
             if not inst.auth(user):
                 raise exceptions.PermissionDenied
-        node.primary_institution = inst
+            node.add_primary_institution(inst=inst, user=user)
```

Comment:
As suggested in https://github.com/CenterForOpenScience/osf.io/pull/5017/files#r53060585, expected exceptions shuold be handled here, e.g. `UserNotAffiliated`.